staging/src/k8s.io/apiserver/pkg/server/openapi/openapi.go

```diff
@@ -64,14 +97,37 @@ func RegisterOpenAPIService(servePath string, webServices []*restful.WebService,
 		return err
 	}
 
-	mux.UnlistedHandleFunc(servePath, func(w http.ResponseWriter, r *http.Request) {
-		resp := restful.NewResponse(w)
-		if r.URL.Path != servePath {
-			resp.WriteErrorString(http.StatusNotFound, "Path not found!")
-		}
-		// TODO: we can cache json string and return it here.
-		resp.WriteAsJson(o.swagger)
-	})
+	mime.AddExtensionType(".json", MIME_JSON)
+	mime.AddExtensionType(".pb", MIME_PB)
+	mime.AddExtensionType(".gz", MIME_PB_GZ)
+
+	type fileInfo struct {
+		ext  string
+		data []byte
+		etag string
+	}
+
+	files := []fileInfo{
+		{".json", o.swaggerBytes, computeEtag(o.swaggerBytes)},
```

Comment:
Could you compute etag in the loop? That would remove the extra field in the struct.